### PR TITLE
Add unit tests for export templates

### PR DESCRIPTION
**Pull request by RevoluPowered**
Partial implementation of https://github.com/godotengine/godot-proposals/issues/1987

### CODE REVIEW

File: modules/gdscript/tests/test_gdscript.cpp [hunk 1]
```diff
@@ -175,8 +175,9 @@ static void test_compiler(const String &p_code, const String &p_script_path, con
 			signature += func->get_argument_name(i);
 		}
 		print_line(signature + ")");
-
+#ifdef TOOLS_ENABLED
 		func->disassemble(p_lines);
```
Comment (RevoluPowered) [inline modules/gdscript/tests/test_gdscript.cpp:179]: can we make this work in tools release?

File: tests/test_text_server.h [hunk 2]
```diff
@@ -247,3 +249,4 @@ TEST_SUITE("[[TextServer]") {
 }; // namespace TestTextServer
 
 #endif // TEST_TEXT_SERVER_H
+#endif // TOOLS_ENABLED
```
Comment (Calinou) [inline tests/test_text_server.h:252]: Add missing newline at end of file:

```suggestion
#endif // TOOLS_ENABLED
```

**Review by Calinou — DISMISSED** [review]

File: tests/test_text_server.h (hunk 2 above)
Comment (RevoluPowered) [reply tests/test_text_server.h]: OK I just git hooked this, should be fixed.

File: modules/gdscript/tests/test_gdscript.cpp (hunk 1 above)
Comment (RevoluPowered) [reply modules/gdscript/tests/test_gdscript.cpp:179]: meh another time

**Comment by RevoluPowered** [comment]
Sorted out sorry it took so long.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**